docs/blog/2019-02-20-getting-started-with-gatsby-themes/index.md

```diff
@@ -0,0 +1,299 @@
+---
+
+### What is a Gatsby theme?
+
+<pullquote cite="Jason Lengstorf">Gatsby themes allow you to focus only on the parts of the site and app building process that you need to care about by abstracting the rest away into a package.</pullquote>
+
+# Getting Started
+
+Create a new directory
+
+`mkdir gatsby-themes`
+
+Navigate to the directory
+
+`cd gatsby-themes`
+
+Create a _package.json_ file
+
+`npm init`
+
+Tidy up your _package.json_ file and create workspaces which includes the project name, site, and your packages. Both of these directories will include their own _package.json_ files.
+
+```json
+{
+  "name": "gatsby-site",
+  "private": true,
+  "version": "1.0.0",
+  "workspaces": ["site", "packages/*"]
+}
+```
+
+Next, you want to create your _site_ directory and your _packages_ directory within your _gatsby-theme_ project directory. Make sure the names that you choose for your directories are the same as what you put in your workspaces. You will also want to go into your packages directory and make another directory with the name of your theme. For the purpose of this tutorial, we will call it _theme_. Then you will want to `yarn init` the _theme_ directory and the _site_ directory.
+
+```
+mkdir site
+mkdir packages
+cd packages
+mkdir theme
+cd theme
+yarn init
+cd ../../site/
+yarn init
+```
+
+After using `yarn init` you will be asked a few questions. The main thing you will want to pay attention to is the entry point (index.js). For the _theme_ directory, you can leave the entry point as _index.js_ (just make sure you have an _index.js_ file) and for the _site_ directory, you can make entry point _gatsby-config.js_. Next, you will want to go into your _site_ directory and do `yarn workspace site add gatsby`.
+
+Hooray! Gatsby should now be added to your site directory if you look in your _package.json_ file. Adjustments you make to your file is adding "scripts" and adding the name of your Gatsby theme, in this case, _theme_, to your dependencies.
+
+```json
+{
+  "name": "theme",
+  "version": "0.0.1",
+  "description": "Practicing making a Gatsby theme",
+  "main": "index.js",
+  "license": "MIT",
+  "devDependencies": {
+    "gatsby": "^2.0.118",
+    "react": "^16.8.1",
+    "react-dom": "^16.8.1"
+  },
+  "peerDependencies": {
+    "gatsby": "^2.0.118",
+    "react": "^16.8.1",
+    "react-dom": "^16.8.1"
+  },
+  "dependencies": {
+    "@mdx-js/mdx": "^0.17.0",
+    "@mdx-js/tag": "^0.17.0",
+    "gatsby-mdx": "^0.3.6",
+    "gatsby-plugin-page-creator": "^2.0.6"
+  }
+}
+```
+
+You will want to add Gatsby, React, and ReactDOM to as dev dependencies for _site_.
+
+`yarn workspace site add gatsby react react-dom -D`
+
+Then you will navigate out of the _site_ directory and add Gatsby, React, and ReactDOM as dev dependencies for _theme_.
+
+`yarn workspace theme add gatsby react react-dom -D`
+
+You will want to make Gatsby, React, and ReactDom peer dependencies in both the _site_ and _theme_ directories.
```

Comment:
We won't need to specify these peer dependencies in the "site" project itself since it isn't being used as a library.